### PR TITLE
Adds OSC52 copy escape sequence support

### PR DESCRIPTION
**Pull request by isundaylee**
Currently I can only get the copied content in the tmux copy buffer, not in the
system clipboard. This commit adds an option to print the copied text as a
OSC52 copy escape sequence, which in supported terminals (tested in iTerm) will
be copied to the system clipboard.

### CODE REVIEW

**Comment by fcsonline** [comment]
Thanks for this contribution! :muscle:

Next days I'm going to test it. I didn't know about all the OSC52 thing :sweat_smile:

File: README.md [hunk 1]
```diff
@@ -248,6 +249,18 @@ For example:
...
+### @thumbs-osc52
+
+`default: 0`
+
+If this is set to `1`, `tmux-thumbs` will print a OSC52 copy escape sequence when you select a match, in addition to running the pick command. This sequence, in terminals that support it (e.g. iTerm), allows the content to be copied into the system clipboard in addition to the tmux copy buffer.
+
+For example:
+
+```
+set -g @thumbs-osc51 1
```
Comment (fcsonline) [inline README.md]: Typo

**Comment by isundaylee** [comment]
@fcsonline Thank you for your time! Also thanks for writing this great plugin :)

**Comment by fcsonline** [comment]
It is not working for me. Should I enable something else in iTerm a part of
_Applications in terminal may access clipboard_?

**Comment by isundaylee** [comment]
@fcsonline Actually I had the same issue when I tested it on my work machine today. Will investigate and fix.

**Comment by fcsonline** [comment]
Ok. And the copy&paste with the uppercase key stopped to work too. I'm not sure if it's related.

**Comment by isundaylee** [comment]
@fcsonline I think I found the reason for both issues. Sorry for accidentally breaking the upcase command. I'll push a fix with more details tonight.

**Comment by isundaylee** [comment]
I pushed a fix for both issues. Let me know if that works. 

Also, let me know if you prefer I squash all commits into one single commit or leave them as is.

**Comment by isundaylee** [comment]
I just rebased on top of latest master.

**Comment by kprav33n** [comment]
This will be quite helpful. Any plan on merging this PR?

**Comment by fcsonline** [comment]
I'm going to try to rebase again from master and test it again.

**Comment by fcsonline** [comment]
I created a new branch with the rebase + some tweaks: https://github.com/fcsonline/tmux-thumbs/compare/osc52 . There is an extra commit with the tweaks. @isundaylee feel free to squash into your commit if you confirm it works with the latest changes.

    Adapt old osc52 implementation to swapper and rustbox era
    
    OSC52 support was implemented before two major refactors in tmux-thubs:
    
    - Decouple tmux-thumbs from tmux
    - Removal of rustbox with termion
    
    I'm not sure if is the right decision, but I moved the arguments to
    swapper land because it looks like more tmux related. If it makes sense
    for other terminals to support this we can move it back.
    
    The question we need to answer here if it makes sense for standalone `thumbs`:
    
    ```
    cat sample/test1 | ./target/release/thumbs --osc52
    ```

**Comment by kprav33n** [comment]
Sweet! Let me test it and get back to you.

> The question we need to answer here if it makes sense for standalone `thumbs`

IMHO, this will be immensely useful. There is no such alternative that I know of. Hence, I just created a simple Rust utility called [oscopy](https://github.com/kprav33n/oscopy) that kind of behaves like `xsel` or `pbcopy`.

**Comment by fcsonline** [comment]
@kprav33n Did you have the opportunity to test the branch?

**Comment by kprav33n** [comment]
I tried building off of isundaylee/tmux-thumbs:osc5. It doesn't copy the to the host clipboard. I am using macOS Catalina (10.15.5) and iTerm 3.3.11.

**Comment by isundaylee** [comment]
Note that in iTerm "Preferences -> General -> Selection -> Applications in terminal may access clipboard" needs to be enabled.

Other than that, I think I had issues on some hosts due to a tmux race condition, which I was able to fix with this commit: https://github.com/isundaylee/tmux-thumbs/commit/2cf07126d479a9402bd4a05273afa78fb1549394

**Comment by sturob** [comment]
Tested and works when using the latest Crostini terminal (running under Linux on a Chromebook).

Just had to add `set -g @thumbs-osc52 1` to my tmux.conf.

Great work!

**Comment by fcsonline** [comment]
As @sturob said, it seems to work. I'm going to merge the branch. If we see something not working properly we can open a new pull request.

**Comment by fcsonline** [comment]
Merged to master! Closing this PR due the conflicts.

**Pull request closed without merging**